src/workerd/api/http.c++

```diff
@@ -1176,7 +1178,41 @@ void Request::shallowCopyHeadersTo(kj::HttpHeaders& out) {
 }
 
 kj::Maybe<kj::String> Request::serializeCfBlobJson(jsg::Lock& js) {
-  return cf.serialize(js);
+  if (cacheMode == CacheMode::NONE) {
+    return cf.serialize(js);
+  }
+
+  int ttl = 2;
+  switch (cacheMode) {
+    case CacheMode::NOSTORE:
+      ttl = -1;
+      break;
+    case CacheMode::NOCACHE:
+      ttl = 0;
+    case CacheMode::NONE:
+      KJ_ASSERT(false);
+  }
+
+  CfProperty clone;
+  KJ_IF_SOME(obj, cf.get(js)) {
+    (void)obj;
+    clone = cf.deepClone(js);
+  } else {
+    clone = CfProperty(js, js.obj());
+  }
+  auto obj = KJ_ASSERT_NONNULL(clone.get(js));
+
+  if (obj.has(js, "cacheTtl")) {
+    jsg::JsValue oldTtl = obj.get(js, "cacheTtl");
+    JSG_REQUIRE(oldTtl == js.num(ttl), TypeError,
+        kj::str("CacheTtl: ", oldTtl, ", is not compatible with cache: ",
+            getCacheModeName(cacheMode).orDefault("none"_kj), " header."));
+  } else {
+    obj.set(js, "cacheTtl", js.num(ttl));
+  }
+
+  obj.set(js, "cf-cache-level", js.str("byc"_kjc));
```

Comment:
This should not be set if `CacheMode::NOCACHE` should it?

Reply:
We aren"t supporting no-cache yet. That's why it there is no break in the switch in that case.